### ISSUE TITLE
Validation via Ajax using prepareForValidation method

### ISSUE THREAD

**Issue by lukasjuhas** [Propaganistas/Laravel-Phone#86]
Hey, thanks for your package.

I'm having issues when validating phone numbers using ajax - in particular Vue.js and axios.

I'm validaitng the phone number using request classes, which works well until I try to do the same (same class) using ajax. 

```
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;
use App\Traits\NormaliseTelephoneTrait;

class TelephoneStoreRequest extends FormRequest
{
    use NormaliseTelephoneTrait;
    /**
     * Determine if the user is authorized to make this request.
     *
     * @return bool
     */
    public function authorize()
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules()
    {
        return [
            'telephone' => [
                'required',
                'max:255',
                'phone:AUTO,MOBILE'
            ],
        ];
    }

    /**
     * Modify the input values
     *
     * @return void
     */
    protected function prepareForValidation()
    {
        // if provided phone number can't be formatted (ie. phone number doesn't exist for countries provided)
        // the library will throw an exception
        try {
            // replace old input with new input
            $this->request->set('telephone', $this->normaliseTelephone($this->get('telephone')));
        } catch (\Exception $e) {
            // don't do anything as the rule will display appropriate error per validation
        }
    }
}

```
The `normaliseTelephone` trait will basically convert a phone number to a desiret format. E.g. 07724726150 will become +447724726150.

When I submit the form without ajax, and I 07724726150, or even in already converted format +447724726150 - validation passes as expected as that's a valid phone number.

Now, If I do exactly the same via ajax, using the same request class, the +447724726150 works and passes but 07724726150 will result in  "The telephone must be a valid mobile phone number".

Now when I tried to debug this, in the `rules()` method, I have `dd`'d the `telephone` field like this: `dd($this->get('telephone'));` I get exactly the same format for both formats of the number although the validation fails trhough ajax.

So I wonder if there is a problem with ajax support or if the `prepareForValidation` method does not work when running validation through ajax? 

Thanks!

**Comment by lukasjuhas** [comment]
Okay, I've found the issue. It is not releated to this package at all. Apologies!

If anyone is interested it waas due to [axios](https://github.com/axios/axios). I did not use x-www-form-urllencoded format.

https://github.com/axios/axios#using-applicationx-www-form-urlencoded-format

Thank you
